### DISCUSSION TITLE
Dec/HA grid and directional elongation values

### DISCUSSION THREAD

**Discussion by sparklingraccoon** [Stellarium/stellarium#2368] [Ideas]
Hey everyone! I'm an astronomy lab class assistant at the university level, and we use Stellarium regularly to teach. In our curriculum, we use dec/HA instead of dec/RA because we've found HA is a bit more intuitive. Especially because our intro class is meant for a nontechnical audience, we try to make things as smooth as possible.

On that note, it would be very useful to have a dec/HA grid in addition to the existing dec/RA one. Additionally, the ability to adjust the step sizes (ex: I want to see lines every 15° instead of every 10°, or every 2h instead of every 1h) would be AWESOME! Especially when projecting onto a larger screen, adjustability is key. 

Also another small tidbit: Elongation values for planets are given as a positive number, no matter which side of the Sun they're on. For introductory students, it would be a lot more intuitive to have the sign mean something (like + for E and - for W) or just display E or W. Especially for planets near opposition, it would be handy to show which direction the elongation is. 

Anyway, thanks for reading! I look forward to seeing these (probably relatively easy) changes implemented! If you have questions, feel free to reach out.

**Comment by alex-w** [comment]
Both ideas are acceptable and we'll implement it later, or, you may help us with it, if you want got it faster.

**Reply by sparklingraccoon** [reply]
Thank you! I love how involved you are in the community. How would I be able to help?

**Comment by gzotti** [comment]
We recently added &Delta;&lambda;<sub>s</sub> as second interpretation of "elongation". Here it is easy to decide upon the sign. The other elongation just means "angular distance from the sun", which is always positive.

**Reply by alex-w** [reply]
@gzotti maybe you have time to the extending output format of the ecliptic elongation?

**Reply by gzotti** [reply]
Not this week, but the idea is worth implementing. Likewise a HA grid which is mostly copy/paste for yet another grid. User-Configurable step width was a previous idea/wishlist proposal but is more work to think about, as the auto-configuration works IMHO pretty well.

**Reply by alex-w** [reply]
> Likewise a HA grid which is mostly copy/paste for yet another grid. 

Why? Just a flags for 2 current grids (a 2 checkboxes in the GUI) to switch between first and second equatorial coordinate systems

**Reply by gzotti** [reply]
I would want to see both at the same time :-)

**Reply by alex-w** [reply]
> I would want to see both at the same time :-)

OK, a tri-state flag - I, II and I+II  :)

But yes, probably it need some refactoring the grids code

**Reply by sparklingraccoon** [reply]
I think for elongation, something handy could be an accesibility option box in the settings, like the one for “Azimuth with respect to South.” It could be default off, but if someone like me wants it (even though it’s technically incorrect), they can just tick a “Directional elongation” box.

**Reply by sparklingraccoon** [reply]
Also with the grids, my intuition was adding the HA grid as a separate option in all of the “Markings.” I think that may be easier for the user to select, rather than having to toggle a flag.

**Reply by gzotti** [reply]
See https://github.com/Stellarium/stellarium/pull/2399
It's however a bit more difficult than anticipated. The new equator (with hour angle marks) is done, but the grid is constructed a bit more difficult.

**Comment by alex-w** [comment]
Please check the latest stable version of Stellarium:
https://github.com/Stellarium/stellarium/releases/latest

**Comment by alex-w** [comment]
Hello @sparklingraccoon! Please check the fresh version (development snapshot) of Stellarium:
https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot
